File: examples/example-dart/lib/app/runner.dart
import '../model/user.dart';
import 'package:sample_app/model/profile.dart';

abstract class BaseRunner {}
mixin Runnable {}

class Runner extends BaseRunner with Runnable {
  String run(User user) {
    return user.name;
  }
}

String boot(Profile profile) {
  return Runner().run(User(profile.name));
}

File: examples/example-lua/app/runner.lua
local user = require("app.model.user")

local Runner = {}

function Runner.run(name)
  return user.new(name)
end

function Runner.greet(name)
  return "Hello " .. Runner.run(name).name
end

local function boot()
  return Runner.run("Ada")
end

return Runner
